### PR TITLE
docs/bootc-install: Give more guidance on `root=` and `rootflags=`

### PR DESCRIPTION
**Pull request by cgwalters**
We need to change Anaconda and bootc-image-builder to stop generating `/etc/fstab` for this reason.

### CODE REVIEW

File: docs/src/bootc-install.md [hunk 1]
```diff
@@ -222,6 +225,39 @@ a different sandboxing tool (e.g. [bubblewrap](https://github.com/containers/bub
...
+
+Systems using systemd will often default to using
+[systemd-fstab-generator](https://www.freedesktop.org/software/systemd/man/latest/systemd-fstab-generator.html)
+and/or [systemd-gpt-auto-generator](https://www.freedesktop.org/software/systemd/man/latest/systemd-gpt-auto-generator.html#).
+Support for the latter though for the root filesystem is conditional on EFI and a bootloader implementing the bootloader interface.
+
+Outside of the discoverable partition model, a common baseline default for installers is to set `root=UUID=`
+(and optionally `rootflags=`) kernel arguments as machine specific state.
+When using `install to-filesystem`, you should provide these as explicit
+kernel arguments.
```
Comment (vrothberg) [inline docs/src/bootc-install.md:244]: Can you give an example?  This stuff is outside my comfort zone, so having some example would increase my understanding.

**Review by vrothberg — COMMENTED** [review]
Is there a workaround to avoid the remount service from failing?

File: docs/src/bootc-install.md (hunk 1 above)
Comment (cgwalters) [reply docs/src/bootc-install.md:244]: This section of the document is targeted at people doing generic installers, disk image image generation tools and custom installs in general.

There's multiple ways to set kernel arguments that are described in the docs already right?

**Comment by cgwalters** [comment]
> Is there a workaround to avoid the remount service from failing?

No, unfortunately.

I commented in https://bugzilla.redhat.com/show_bug.cgi?id=2332319#c1 which should cover the Anaconda case you hit. We can link that as a known bug until Anaconda is updated.

The change that would be by far the easiest/biggest impact would be to change the systemd fstab generator to handle this but I've hesitated to raise it there as I think conceptually it's more that the fstab is wrong.

**Comment by vrothberg** [comment]
> I commented in https://bugzilla.redhat.com/show_bug.cgi?id=2332319#c1 which should cover the Anaconda case you hit. We can link that as a known bug until Anaconda is updated.

Love that!

**Review by vrothberg — APPROVED** [review]

**Pull request merged**